### ISSUE TITLE
508-defect-1 [COGNITION]: BAM2 Debt Letters - Focus must be managed when routes changes to debt detail, debt letter downloads

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#16776]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Jennifer_

## User Story or Problem Statement

As a screen reader user, I want to know when routes change and the debt detail or download debt letters views have loaded. Right now I do not hear anything to tell me the action I took has completed or failed.

## Details

The Debt Detail view and Download Debt Letters view do not manage focus dynamically. This means screen readers do not have a cue to announce the page has loaded for users. Given these pages have to fetch data to render, we need to set focus on the H1 when the page is ready.

## Acceptance Criteria

- [ ] Focus is set on the H1 successfully on page load (may need to hook into `componentDidMount()` lifecycle method)
- [ ] Screen readers announce the route change (NVDA, JAWS, VoiceOver, iOS VoiceOver)

## Environment

* https://staging.va.gov/manage-va-debt/your-debt/debt-detail
* https://staging.va.gov/manage-va-debt/your-debt/debt-letters
Test users 1, 38

## Steps to Recreate

1. Log in with test user
1. Start a screen reader of your choosing
1. Load the https://staging.va.gov/manage-va-debt/your-debt view
1. Click a debt detail or the download debt letters link
1. Verify nothing is announced when the app view renders

## Solution (if known)

```diff
! Need to add the tabindex="-1" to the H1 and a JavaScript method to focus it on page ready

<h1
  id="downloadDebtLetters"
  class="vads-u-margin-bottom--2"
+ tabindex="-1"
>Download debt letters </h1>
```

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

![Screen Shot 2020-12-01 at 3 55 22 PM](https://user-images.githubusercontent.com/934879/100802468-2b349a00-33ef-11eb-98c4-1acc47760b52.png)

---
![Screen Shot 2020-12-01 at 3 55 32 PM](https://user-images.githubusercontent.com/934879/100802492-312a7b00-33ef-11eb-89fd-0d9aa6040b00.png)

**Comment by 1Copenut** [comment]
@leahkeeler I had to bump this one to a launch blocker. The two app pages for debt detail and debt history need to announce themselves to screen reader users. This is a pattern we use in a lot of other applications, so please reach out to Jennifer if you have any questions.

**Comment by bradl3yC** [comment]
@1Copenut This PR was just merged and should be available for validation in staging in the next hour

**Comment by 1Copenut** [comment]
@bradl3yC This is functioning as expected. I'd like to have the yellow halo removed from the H1, and we can close this issue when that's done. We're providing guidance not to set the focus halo on non-native elements like headings because they're not expected to receive focus normally. JAWS and VoiceOver provide their own virtual cursor halo around the current element, and NVDA has a plugin to do this as well.

---
<img width="784" alt="Screen Shot 2020-12-03 at 10 35 56 AM" src="https://user-images.githubusercontent.com/934879/101059214-a1eaa800-3553-11eb-8132-cb4edb35e6ef.png">

---
<img width="1104" alt="Screen Shot 2020-12-03 at 10 33 56 AM" src="https://user-images.githubusercontent.com/934879/101059238-a911b600-3553-11eb-8f0f-fbd459547f57.png">

**Comment by bradl3yC** [comment]
@1Copenut whats the appropriate way to set the focus then? a tabIndex of -1 did not work for this

**Comment by 1Copenut** [comment]
@bradl3yC Focus is working as expected. I'm just looking for the styling to be adjusted to remove the yellow focus halo on the H1. This code is not prod-ready, just for illustration:

```css
h1 {
  outline: 0;
}
```

We use this technique with the `tabindex="-1"` to set focus with JavaScript, and not show the visual indicator.

**Comment by bradl3yC** [comment]
@1Copenut Ill change the CSS - when I set the tabIndex it didn't set the focus, and the tab order went all over the page and then back to the heading - it didnt strike me as desirable outcome

**Comment by 1Copenut** [comment]
@bradl3yC I should have better described the behavior. You should keep the `tabindex="-1"` on the H1, and call a `setFocus()` method on component mount. We use this technique in a lot of applications, I'll find a few in staging quick and update my comment.

**Comment by bradl3yC** [comment]
@1Copenut I left the tabIndex in :)

**Comment by bradl3yC** [comment]
@1Copenut this is merged

**Comment by 1Copenut** [comment]
Spot checked with MacOS + Safari + VoiceOver and Win10 + Chrome + JAWS/NVDA. All worked as expected. Moving to close.

/cc @leahkeeler & @bradl3yC

**Comment by leahkeeler** [comment]
Thank-you Sir :) Hate having those launch-blockers
